src/thunderbots/software/util/ros_messages.cpp

```diff
@@ -180,5 +178,62 @@ namespace Util
             World world(field, ball, friendly_team, enemy_team);
             return world;
         }
+
+        thunderbots_msgs::World transformWorldMessage(
+            const thunderbots_msgs::World& old_world_msg)
+        {
+            thunderbots_msgs::World new_world_msg;
+
+            new_world_msg.ball = transformBallMessage(old_world_msg.ball);
+            new_world_msg.friendly_team.robots =
+                transformRobotMessages(old_world_msg.friendly_team.robots);
+            new_world_msg.enemy_team.robots =
+                transformRobotMessages(old_world_msg.enemy_team.robots);
+
+            return new_world_msg;
+        }
+
+        thunderbots_msgs::Ball transformBallMessage(
+            const thunderbots_msgs::Ball& old_ball_msg)
+        {
+            thunderbots_msgs::Ball new_ball = old_ball_msg;
```

Comment:
make the `new_ball_msg` a copy of the `old_ball_msg` initially, so we only transform and overwrite data we need to. Otherwise the new `ball_msg` we return is missing information.